File: Microservice_Rate_Limiter_Logic.py
def obtener_limite_peticiones(tipo_usuario, en_mantenimiento):
    
    if en_mantenimiento:
        return 0  
    if tipo_usuario == "Premium":
        return 1000
    elif tipo_usuario == "Standard":
        return 100
    else:
        print(f"Tipo de usuario '{tipo_usuario}' no reconocido. Límite mínimo aplicado.")
        return 0

print(f"Usuario Premium (Activo): {obtener_limite_peticiones('Premium', False)} req/s")
print(f"Usuario Standard (Activo): {obtener_limite_peticiones('Standard', False)} req/s")
print(f"Sistema en Mantenimiento: {obtener_limite_peticiones('Premium', True)} req/s")

File: MicroserviceRateLimiterLogic.js
function obtenerLimitePeticiones(tipoUsuario, enMantenimiento) {
    console.log(`\n--- Verificando acceso para: ${tipoUsuario} ---`);

    if (enMantenimiento) {
        console.log("ALERTA: Servidor en Mantenimiento. Acceso denegado a todos.");
        return 0;
    }

    let limite = 0;

    if (tipoUsuario === "Premium") {
        limite = 1000;
        console.log(`Usuario Premium detectado. Límite: ${limite} req/s.`);
    } 
    else if (tipoUsuario === "Standard") {
        limite = 100;
        console.log(`Usuario Standard detectado. Límite: ${limite} req/s.`);
    } 
    else {
        limite = 0;
        console.log("Tipo de usuario desconocido. Límite denegado.");
    }

    return limite;
}

obtenerLimitePeticiones("Premium", false);
obtenerLimitePeticiones("Standard", false); 
obtenerLimitePeticiones("Premium", true);